airbyte-integrations/connectors/source-salesforce/unit_tests/conftest.py

```diff
@@ -98,37 +96,45 @@ def _stream_api(stream_config, describe_response_data=None):
 
 @pytest.fixture(scope="module")
 def stream_api(stream_config):
-    return _stream_api(stream_config)
+    return mock_stream_api(stream_config)
 
 
 @pytest.fixture(scope="module")
 def stream_api_v2(stream_config):
     describe_response_data = {"fields": [{"name": "LastModifiedDate", "type": "string"}, {"name": "BillingAddress", "type": "address"}]}
-    return _stream_api(stream_config, describe_response_data=describe_response_data)
+    return mock_stream_api(stream_config, describe_response_data=describe_response_data)
 
 
 @pytest.fixture(scope="module")
 def stream_api_pk(stream_config):
     describe_response_data = {"fields": [{"name": "LastModifiedDate", "type": "string"}, {"name": "Id", "type": "string"}]}
-    return _stream_api(stream_config, describe_response_data=describe_response_data)
+    return mock_stream_api(stream_config, describe_response_data=describe_response_data)
 
 
 @pytest.fixture(scope="module")
 def stream_api_v2_too_many_properties(stream_config):
     describe_response_data = {"fields": [{"name": f"Property{str(i)}", "type": "string"} for i in range(Salesforce.REQUEST_SIZE_LIMITS)]}
     describe_response_data["fields"].extend([{"name": "BillingAddress", "type": "address"}])
-    return _stream_api(stream_config, describe_response_data=describe_response_data)
+    return mock_stream_api(stream_config, describe_response_data=describe_response_data)
 
 
 @pytest.fixture(scope="module")
 def stream_api_v2_pk_too_many_properties(stream_config):
     describe_response_data = {"fields": [{"name": f"Property{str(i)}", "type": "string"} for i in range(Salesforce.REQUEST_SIZE_LIMITS)]}
     describe_response_data["fields"].extend([{"name": "BillingAddress", "type": "address"}, {"name": "Id", "type": "string"}])
-    return _stream_api(stream_config, describe_response_data=describe_response_data)
+    return mock_stream_api(stream_config, describe_response_data=describe_response_data)
+
 
+def generate_stream(stream_name, stream_config, stream_api, state=None, legacy=True):
+    if state is None:
+        state = _ANY_STATE
 
-def generate_stream(stream_name, stream_config, stream_api):
-    return SourceSalesforce.generate_streams(stream_config, {stream_name: None}, stream_api)[0]
+    stream = SourceSalesforce(_ANY_CATALOG, _ANY_CONFIG, state).generate_streams(stream_config, {stream_name: None}, stream_api)[0]
+    if legacy and hasattr(stream, "_legacy_stream"):
```

Comment:
This is another thing I don't like. We will start migrating those tests at some point